libc: replace musl's and MinGW's trigonometric functions with compiler_rt's

papparapa opened the pull request:
Part of #2879.
According to the [instruction](https://github.com/ziglang/zig/issues/2879#issuecomment-2797310223), this PR removes some musl files, including the following functions:

- musl
    - sin
    - sinf
    - cos
    - cosf
    - sincos
    - sincosf
    - tan
    - tanf
- MinGW (added after review)
    - sinf
    - cosf
    - sincos
    - sincosf
    - tanf

I confirmed that running `zig build test-modules` and [zig-libc-test](https://github.com/rpkak/zig-libc-test) raised no errors during the trigonometric function tests.
zig-libc-test revealed some discrepancies between the results of long double trigonometric functions in compiler_rt and musl. Therefore, this PR is restricted to float and double functions.

alexrp commented:
It seems like we also have

* `cosf`
* `sincos`
* `sincosf`
* `sinf`
* `tanf`

in `lib/libc/mingw`.

papparapa commented:
When deleting musl files, I was able to run the deletion after confirming with zig-libc-test that the function behavior did not change.
However, I am not sure what policy I can use to delete MinGW files.
The behavior of compiler_rt's `sin` function and musl's `sin` function are consistent, but MinGW's `sin` function may not be.
Do I need to make sure that the behavior of compiler_rt's `sin` function and MinGW's `sin` function match?
Or, even if the behavior does not match, can we assume that musl's `sin` function (compiler_rt's `sin` function) behavior is more appropriate and remove MinGW's `sin` function?

papparapa commented:
Thanks, I deleted the MinGW files with relief. (71ff3830)
In `lib/libc/mingw/math/arm/sincos.S` and `/lib/libc/mingw/math/arm64/sincos.S`, I removed only the `sincos` symbol.
It will not be too late to remove MinGW's `sincosl` at the same time as musl.

alexrp commented on lib/libc/mingw/math/arm/sincos.S:
```diff
@@ -8,11 +8,8 @@
 	.file	"sincos.S"
 	.text
 	.align 2
-	.globl __MINGW_USYMBOL(sincos)
```
There's a high likelihood of this change being lost in a future MinGW-w64 source import. To prevent that, add a line like this:

```asm
/* zig patch: remove sincos symbol */
```

papparapa commented on lib/libc/mingw/math/arm64/sincos.S:
```diff
@@ -8,11 +8,8 @@
 	.file	"sincos.S"
 	.text
 	.align 2
-	.globl __MINGW_USYMBOL(sincos)
```
Added a comment (7bc3f294)

andrewrk commented:
Nice, thank you
